### PR TITLE
Add staticWhich around most executable paths in obelisk-command

### PR DESCRIPTION
**Pull request by lpsmith**
I have:

  - [x] Based work on latest `develop` branch
  - [x] Looked for lint in my changes with `hlint .` (lint found code you did not write can be left alone)
  - [x] Run the test suite: `$(nix-build -A selftest --no-out-link)`
  - [ ] (Optional) Run CI tests locally: `nix-build release.nix -A build.x86_64-linux --no-out-link` (or `x86_64-darwin` on macOS)

### CODE REVIEW

File: lib/command/src/Obelisk/Command/Deploy.hs [hunk 1]
```diff
@@ -182,9 +184,9 @@ deployPush deployPath getNixBuilders = do
   isClean <- checkGitCleanStatus deployPath True
   when (not isClean) $ do
     withSpinner "Committing changes to Git" $ do
-      callProcessAndLogOutput (Debug, Error) $ proc "git"
+      callProcessAndLogOutput (Debug, Error) $ proc gitPath
         ["-C", deployPath, "add", "."]
-      callProcessAndLogOutput (Debug, Error) $ proc "git"
+      callProcessAndLogOutput (Debug, Error) $ proc gitPath
```
Comment (3noch) [inline lib/command/src/Obelisk/Command/Deploy.hs:189]: This should use `gitProc`

File: lib/command/src/Obelisk/Command/VmBuilder.hs [hunk 2]
```diff
@@ -98,12 +98,12 @@ setupNixDocker stateDir = withSpinner ("Creating Docker container named " <> con
...
   callProcessAndLogOutput (Debug, Error) $
-    proc "ssh-keygen" ["-t", "ed25519", "-f", stateDir </> sshKeyFileName, "-P", ""]
+    proc sshKeygenPath ["-t", "ed25519", "-f", stateDir </> sshKeyFileName, "-P", ""]
 
   -- Build the docker container (which uses the SSH keys in the 'ssh' folder)
   containerId <- fmap T.strip $ readProcessAndLogStderr Error $
-    proc "docker" ["build", stateDir, "--quiet"]
-  callProcessAndLogOutput (Debug, Error) $ proc "docker"
+    proc dockerPath ["build", stateDir, "--quiet"]
+  callProcessAndLogOutput (Debug, Error) $ proc dockerPath
```
Comment (3noch) [inline lib/command/src/Obelisk/Command/VmBuilder.hs:106]: Let's leave docker out of this for now. None of these commands will work unless they have docker installed anyway and this is only for mac.

Comment (lpsmith) [reply lib/command/src/Obelisk/Command/VmBuilder.hs:106]: docker has been left out of it.

File: lib/command/src/Obelisk/Command/Deploy.hs (hunk 1 above)
Comment (lpsmith) [reply lib/command/src/Obelisk/Command/Deploy.hs]: Fixed

File: lib/command/src/Obelisk/Command/VmBuilder.hs (hunk 2 above)
Comment (3noch) [reply lib/command/src/Obelisk/Command/VmBuilder.hs:106]: Excellent thanks.

File: lib/command/src/Obelisk/Command/Utils.hs [hunk 3]
```diff
@@ -39,6 +39,9 @@ mvPath = $(staticWhich "mv")
 rmPath :: FilePath
 rmPath = $(staticWhich "rm")
 
+mkdirPath :: FilePath
+mkdirPath = $(staticWhich "mkdir")
```
Comment (3noch) [inline lib/command/src/Obelisk/Command/Utils.hs:43]: Hm. Why wouldn't we just use haskell to do this ...

Comment (lpsmith) [reply lib/command/src/Obelisk/Command/Utils.hs:43]: Well, we'd have to emulate the `mkdir -p` option in this case, as the `mkdir` command line utility is slightly more than a thin wrapper around the `mkdir(2)` syscall.

That's very doable, but would have to be done.  And I didn't really treat this PR as more than a mechanical transformation + should I apply said mechanical transformation in a particular context.

Honestly, I didn't consider _not_ `staticWhich`'ing `mkdir` in this case.  `which` also was a pretty easy call.  I debated briefly regarding `rsync`, `git`, `ssh`, and `ssh-agent`.

Comment (lpsmith) [reply lib/command/src/Obelisk/Command/Utils.hs:43]: Oh, I forgot `System.Directory` already has emulation for the `-p` option.

Comment (lpsmith) [reply lib/command/src/Obelisk/Command/Utils.hs]: Done

**Pull request merged**